packages/analytics/src/functions.ts

```diff
@@ -41,7 +57,12 @@ export async function logEvent(
     gtagFunction(GtagCommand.EVENT, eventName, eventParams);
     return;
   } else {
-    const measurementId = await initializationPromise;
+    const measurementId = await handleInitializationPromiseErrors(
```

Comment:
I wasn't totally sure what you meant but I tried to do this.

Reply:
Sorry I was not clear. The code looks like that we are doing no-op because measurementId is not available, but in fact it could be that it is available, but the environment check failed, so it is confusing to readers. I think we should make it clear in the code that we do no-op when either environment check fails or measurementId is not available.